### PR TITLE
Add `Command.cooldown`

### PR DESCRIPTION
**Pull request by ImNimboss**
## Summary

This PR is to add a `command.cooldown` property to commands.

Currently the cooldown can only be retrieved by `command._buckets._cooldown` which is, frankly, poking through the internals of the library. A `command.cooldown` property would be preferred.

A use case could be showing the cooldown of a command in a help command.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/core.py [hunk 1]
```diff
@@ -657,6 +660,10 @@ def clean_params(self) -> Dict[str, inspect.Parameter]:
         """
         return self.params.copy()
 
+    @property
+    def cooldown(self) -> Optional[Cooldown]:
+        return self._buckets._cooldown
```
Comment (Rapptz) [inline discord/ext/commands/core.py:665]: Documentation should be moved here.

It should also have `.. versionadded:: 2.0`.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/ext/commands/core.py (hunk 1 above)
Comment (ImNimboss) [reply discord/ext/commands/core.py:668]: Done, sorry for the delay.

File: discord/ext/commands/core.py [hunk 2]
```diff
@@ -657,6 +657,16 @@ def clean_params(self) -> Dict[str, inspect.Parameter]:
         """
         return self.params.copy()
 
+    @property
+    def cooldown(self) -> Optional[Cooldown]:
+        """Optional[:class:`Cooldown`]:
+        The cooldown of a command when invoked or ``None`` if the command
+        doesn't have a registered cooldown.
```
Comment (Rapptz) [inline discord/ext/commands/core.py:664]:
```suggestion
        """Optional[:class:`Cooldown`]: The cooldown of a command when invoked or ``None`` if the command
        doesn't have a registered cooldown.
```

File: discord/ext/commands/core.py [hunk 3]
```diff
@@ -657,6 +657,15 @@ def clean_params(self) -> Dict[str, inspect.Parameter]:
         """
         return self.params.copy()
 
+    @property
+    def cooldown(self) -> Optional[Cooldown]:
+        """Optional[:class:`Cooldown`]: The cooldown of a command when invoked or ``None`` if the command
+        doesn't have a registered cooldown.
+
```
Comment (Rapptz) [inline discord/ext/commands/core.py:664]: This has trailing whitespace which fails black.

Comment (ImNimboss) [reply discord/ext/commands/core.py:664]: Right, sorry. Should be fine to merge now?

Comment (Rapptz) [reply discord/ext/commands/core.py:664]: The trailing whitespace is still there.

Comment (ImNimboss) [reply discord/ext/commands/core.py]: Fixed

**Comment by ImNimboss** [comment]
@Rapptz Following up, will this be merged?

**Comment by Rapptz** [comment]
It's not going to be merged if you can't make it pass CI.

**Comment by Rapptz** [comment]
Now you've reformatted without applying the actual settings used in the repository.

**Comment by ImNimboss** [comment]
I'll just open another PR when I figure out how to contribute properly

**Comment by Rapptz** [comment]
I'll assist you in the discord.py server.

**Pull request closed without merging**